developer-docs-site/docs/tutorials/full-node/network-identity-fullnode.md

```diff
@@ -0,0 +1,174 @@
+      --key-file /path/to/private-key.txt \
+      --output-file /path/to/peer-info.yaml
+  ```
+
+This will create a YAML file that will have your public identity in it. This is useful if you want to connect your FullNode to a specific upstream FullNode, and that FullNode only allows known identities to connect to them.
+
+Example output `peer-info.yaml`:
+
+ ```
+  ---
+  ca3579457555c80fc7bb39964eb298c414fd60f81a2f8eedb0244ec07a26e575:
+    addresses: []
+    keys:
+      - ca3579457555c80fc7bb39964eb298c414fd60f81a2f8eedb0244ec07a26e575
+    role: Downstream
+  ```
+
+In this example, `ca3579457555c80fc7bb39964eb298c414fd60f81a2f8eedb0244ec07a26e575` is the peer ID as well as the public key, which is derived from the private key you generated from the previous step.
+
+
+## Start a node with a static network identity
+
+Once you have the static identity you can startup the FullNode by modifying the configuration file `public_full_node.yaml`:
+
+```
+full_node_networks:
+- network_id: "public"
+  discovery_method: "onchain"
+  identity:
+    type: "from_config"
+    key: "<PRIVATE_KEY>"
+    peer_id: "<PEER_ID>"
+```
+
+In our example, we'd specify:
+
+```
+full_node_networks:
+- network_id: "public"
+  discovery_method: "onchain"
+  identity:
+    type: "from_config"
+    key: "B8BD811A91D8E6E0C6DAC991009F189337378760B55F3AD05580235325615C74"
+    peer_id: "ca3579457555c80fc7bb39964eb298c414fd60f81a2f8eedb0244ec07a26e575"
+```
+
+## Allowing other FullNodes to connect
+
+Once you start your FullNode with a static identity you can allow others to connect to devnet through your node. Follow these recommendations:
+
+- Make sure you open port `6180` (or `6182`, depending on which port your node is listening to) and that you open your firewall.
+- If you are using Docker, simply add `- "6180:6180"` or `- "6182:6182"` under ports in your ``docker-compose.yaml`` file.
+- You'll need to share your FullNode info for others to use as `seeds` in their configurations (e.g., `peer-info.yaml`):
+
+  ```
+  <Peer_ID>:
+    addresses:
+    # with DNS
+    - "/dns4/<DNS_Name>/tcp/<Port_Number>/ln-noise-ik/<Public_Key>/ln-handshake/0"
+    role: Upstream
+  <Peer_ID>:
+    addresses:
+    # with IP
+    - "/ip4/<IP_Address>/tcp/<Port_Number>/ln-noise-ik/<Public_Key>/ln-handshake/0"
+    role: Upstream
+  ```
+
+- Make sure the port number you put in the `addressses` matches the one you have in the fullnode config (`6180` or `6182`). For example:
+
+  ```
+  bb14af025d226288a3488b4433cf5cb54d6a710365a2d95ac6ffbd9b9198a86a:
+    addresses:
+    - "/dns4/pfn0.node.devnet.aptoslabs.com/tcp/6182/ln-noise-ik/bb14af025d226288a3488b4433cf5cb54d6a710365a2d95ac6ffbd9b9198a86a/ln-handshake/0"
+    role: "Upstream"
+  bb14af025d226288a3488b4433cf5cb54d6a710365a2d95ac6ffbd9b9198a86a:
+    addresses:
+    - "/ip4/100.20.221.187/tcp/6182/ln-noise-ik/bb14af025d226288a3488b4433cf5cb54d6a710365a2d95ac6ffbd9b9198a86a/ln-handshake/0"
+    role: "Upstream"
+  ```
```

Comment:
Maybe include a link to https://github.com/aptos-labs/aptos-core/blob/main/documentation/specifications/network/network-address.md with something like "if you want to know more about how `NetworkAddress` is constructed, read this.